src/pages/PeoplePage.tsx

```diff
@@ -0,0 +1,125 @@
+import React, {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { useSearchParams } from 'react-router-dom';
+import { PeopleFilters } from '../components/PeopleFilters';
+import { Loader } from '../components/Loader';
+import { PeopleTable } from '../components/PeopleTable';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { getSearchWith } from '../utils/searchHelper';
+import { getVisiblePeople } from '../helpers/getVisiblePeople';
+import { getPreparedPeople } from '../helpers/getPreparedPeople';
+
+export const PeoplePage: React.FC = React.memo(() => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isDataLoaded, setIsDataLoaded] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const loadPeople = async () => {
+    setIsLoading(true);
+
+    try {
+      const loadedPeople = await getPeople();
+
+      const preparedPeople = getPreparedPeople(loadedPeople);
+
+      setPeople(preparedPeople);
+      setIsDataLoaded(true);
+    } catch (error) {
+      setErrorMessage('Can\'t load people');
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    loadPeople();
+  }, []);
+
+  const sex = searchParams.get('sex');
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries');
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+
```

Comment:
Also, you could use custom hook which takes searchParams object and returns all these fields in the object

Reply:
done